### ISSUE TITLE
[Request] SDXL & Bark

### ISSUE THREAD

**Issue by developerfyp** [Telosnex/fonnx#2]
I want you to make an example, stable diffusion and text to speech bark, thank you, sorry for the trouble

**Comment by jpohhhh** [comment]
TL;DR: Unfortunately, those will both be time-consuming, I'm focused on getting a GPT wrapper app ready by end of 2023. SD XL is too slow on mobile to benefit me, and Bark / TTS requires porting a huge and difficult C library.

**Premier models are more complicated, and model inputs sometimes require implementing a bunch of code**
You can get an ONNX model _fairly_ easily from any model using [HuggingFace doc](https://huggingface.co/docs/transformers/serialization).

Then you have FONNX to run it.

But with big models, like SD XL, you'll run into things like it's actually N separate models. - IIRC SD XL is 4. 

Any given model is a function with some inputs and some outputs. But each model has its own input and outputs that are rarely clean, simple, types

For example, the MiniLM models in this repo. At a high level, they're f(String) => 384 doubles. But in practice, they're f(List<int> inputList) => [[int]], where the inner arrays are all length 384, and the outer array is of length inputList.length.*

So now you need a tokenizer that can convert Strings to tokens.

Multiply that by N models, and I'm using this to build a mobile app, and I don't think I'll be doing SD XL anytime soon.

**Bark / TTS**
I was originally going to close this issue by doing a TTS model. There's a branch [here](https://github.com/Telosnex/fonnx/tree/piper_wip). I found it ~every TTS model depends on eSpeak NG, a C++ library. 

Translating it to Dart is _hard_, many global variables and at least 35K lines of code.
I could instead FFI it, but then, it is of no use to me, I'm targeting web.
Maybe I could WASM it, but that's gotten _harder_ in Flutter over the past couple years, I haven't been able to do it recently. Additionally, the code relies on access to structs serialized to a folder, which will add another layer of complication.

* if you go check, this isn't _actually_ true for the models in this repo: I reworked the models a bit so the averaging of the inputList.length is done in the model itself.
